compiler/passes/normalize.cpp

```diff
@@ -1873,14 +1883,15 @@ static void normVarTypeInference(DefExpr* defExpr) {
         argExpr->baseExpr->replace(new UnresolvedSymExpr("init"));
 
         // Add _mt and _this (insert at head in reverse order)
-        argExpr->insertAtHead(var);
+        argExpr->insertAtHead(new NamedExpr("this", new SymExpr(var)));
```

Comment:
The "this" argument is treated specially, since it starts as generic and its concrete instantiation cannot be known until Phase 1 of the initializer body has been resolved.  Using the NamedExpr on the actual allows us to update the type of the actual once we know its instantiation without relying on the more brittle "it's position 2 in the argument list" signaling.